climada/hazard/low_flow.py

```diff
@@ -203,70 +203,119 @@ def set_from_nc(self, input_dir=None, centroids=None, countries=[], reg=None,
             centr_handling = 'full_hazard'
 
         # read data and call preprocessing routine:
-        self.data, centroids_import = _data_preprocessing_percentile(
+        self.data, centroids_import = data_preprocessing_percentile(
             percentile, yearrange, yearrange_ref, input_dir, gh_model, cl_model,
             scenario, scenario_ref, soc, soc_ref, fn_str_var, bbox, min_days_per_month,
-            keep_dis_data, yearchunks)
+            keep_dis_data, yearchunks, mask_threshold)
 
         if centr_handling == 'full_hazard':
             centroids = centroids_import
         self.identify_clusters()
 
-        # sum "dis" (days per month below threshold) per pixel and cluster_id
-        # and write to hazard.intensiy
+        self.make_climada_hazard(centroids, min_intensity, min_number_cells,
+                                 yearrange, yearrange_ref, gh_model, cl_model,
+                                 scenario, scenario_ref, soc, soc_ref, fn_str_var, keep_dis_data)
+
+    def make_climada_hazard(self, centroids=None, min_intensity=1, min_number_cells=1,
+                            yearrange=TARGET_YEARRANGE, yearrange_ref=REFERENCE_YEARRANGE,
+                            gh_model=None, cl_model=None,
+                            scenario='historical', scenario_ref='historical', soc='histsoc',
+                            soc_ref='histsoc', fn_str_var=FN_STR_VAR, keep_dis_data=False):
+        """ Build low flow hazards with events from clustering and centroids and add attributes.
+        """
+        # sum "dis" (days per month below threshold) per pixel and
+        # cluster_id and write to hazard.intensity
         self.events_from_clusters(centroids)
+
         if min_intensity > 1 or min_number_cells > 1:
             haz_tmp = self.filter_events(min_intensity=min_intensity,
                                          min_number_cells=min_number_cells)
             LOGGER.info('Filtering events: %i events remaining', haz_tmp.size)
             self.event_id = haz_tmp.event_id
             self.event_name = list(map(str, self.event_id))
             self.date = haz_tmp.date
+            self.date_start = haz_tmp.date_start
             self.date_end = haz_tmp.date_end
             self.orig = haz_tmp.orig
             self.frequency = haz_tmp.frequency
             self.intensity = haz_tmp.intensity
             self.fraction = haz_tmp.fraction
             del haz_tmp
         if not keep_dis_data:
-            del self.data
+            self.data = None
         self.set_frequency(yearrange=yearrange)
         self.tag = TagHazard(haz_type=HAZ_TYPE, file_name=\
-                             FILENAME_NC % (gh_model, cl_model, "*", scenario, soc, \
-                                            fn_str_var, "*_*.nc"), \
+                             f'{gh_model}_{cl_model}_*_{scenario}_{soc}_{fn_str_var}_*.nc', \
                              description='yearrange: %i-%i (%s, %s), reference: %i-%i (%s, %s)' \
```

Comment:
This part could probably also be updated to be more readable (line 237/249)

Reply:
done: https://github.com/CLIMADA-project/climada_python/commit/9da58f7626072a15fa20ac1948af8d4d8ace8d19